api/include/opentelemetry/context/context.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include "opentelemetry/common/attribute_value.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/key_value_iterable_view.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+
+// The context class provides a context identifier.
+// This is a dummy class that is meant to be overridden,
+// the methods return default values.
+class Context
+{
+
+public:
+  // The Key class is used to obscure access from the
+  //  user to the context map. The identifier is used as a key
+  // to the context map.
+  class Key
```

Comment:
I'm a bit lost as to what is the real purpose of the Key class. Why not using `nostd::string_view` instead? There's a recent commit that is gonna add ability to use `nostd::string_view` in both `std::map` and `std::unordered_map` . Do I always have to use the `Context::Key` if I need to populate some value in the context? This looks a bit unnatural to me.

Reply:
For example, OT Java SDK - decided that they will convert keys to simple string, very recent commit: https://github.com/open-telemetry/opentelemetry-java/commit/24b4be90c31f8213a50da3aa7a7af94a50891e8f - from a practical standpoint it is much easier to deal with "standard" (or close to standard) primitive than creating a custom Key class.

Reply:
> Why not using nostd::string_view instead?

Sam's not opinionated on the issue, so we'll need a concrete recommendation from a maintainer.

I suggested a Key object because the spec says "The API MUST return an opaque object representing the newly created key.", and be setup such that "Multiple calls to [CreateKey](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/context.md#create-a-key) with the same name SHOULD NOT return the same value".  If we use a string as the key, clients are going to simply use the key name as the key itself, violating that requirement.

Reply:
@0x4b - This is bad for performance. See: https://github.com/open-telemetry/opentelemetry-java/issues/1314

I'd suggest we follow more closely Java implementation than Python or Go, as C++ structurally closer to Java. The issue of Context keys brought up here for Java: https://github.com/open-telemetry/opentelemetry-java/issues/1334 , and spec authors signed off on it allowing String as key for CorrelationContext. I think it'd be the right solution for C++ as well, at least from performance standpoint.

What is your opinion on those two commits above?

Reply:
If this this is bad for performance in Java, it doesn't extrapolate to C++.  In C++ a struct containing a string has no storage or copy overhead compared to the string itself. If there's a concern that string is too expensive, the key can be built around a string_view.

Reply:
@0x4b - we can probably use type alias for the Key class to `nostd::string_view`. Any complex application, even with thousands events has a limited number of known beforehand context keys and event attributes. Initializing these from a string literal as `nostd::string_view` does not require a copy, as storage of these keys does not go out of scope. Recordable or Exporter dealing with async processing of incoming spans / events can perform a memcopy as-needed.

Reply:
@reyang I am removing the Key Object as we spoke about, but I wanted to make sure I did not misinterpret our conversation on Friday and that using a string as a key is the way we want to go.